httpx/cert_reload.go

```diff
@@ -0,0 +1,159 @@
+package httpx
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"path"
+	"runtime"
+	"sync"
+	"time"
+
+	"github.com/fsnotify/fsnotify"
+	"github.com/ory/x/logrusx"
+)
+
+type (
+	CertLocation struct {
+		CertPath string
+		KeyPath  string
+	}
+
+	CertLoadFunc func() ([]tls.Certificate, error)
+
+	CertProvider struct {
+		certs   []tls.Certificate
+		mu      sync.Mutex
```

Comment:
Maybe premature optimization, but this should probably be a sync/Atomic or at least an RWMutex.